faw/pdf-observatory/faw_analysis_set_parse.py

```diff
@@ -1,24 +1,27 @@
 """Manages parsing within analysis sets.
 """
 
-import faw_analysis_set_util
-import faw_internal_util
-import faw_pipelines_util
-
-import dask.distributed
 import io
-import json
+import threading
+import dask.distributed
 import os
+import pathlib
 import psutil
 import pymongo
+import pymongo.collection
 import re
-import shutil
 import signal
 import subprocess
 import tempfile
-import threading
 import time
 import traceback
+from typing import Dict, Any, Optional, Set
+
+import faw_analysis_set_util
```

Comment:
I've left this as-is for now since we discussed it outside of this PR (and since we don't hack any external packages here). Can still change it if you have strong feelings.